tests/integration/tests/test_cloudbeat_integration.py

```diff
@@ -55,6 +55,55 @@ def test_elastic_index_exists(elastic_client, match_type):
     :param match_type: Findings type for matching
     :return:
     """
+    query, sort = build_es_query(match_type)
+    start_time = time.time()
+    result = {}
+    while time.time() - start_time < CONFIG_TIMEOUT:
+        current_result = elastic_client.get_index_data(index_name=elastic_client.index,
+                                                       query=query,
+                                                       sort=sort)
+        if elastic_client.get_total_value(data=current_result) != 0:
+            allure.attach(json.dumps(elastic_client.get_doc_source(data=current_result),
+                                     indent=4,
+                                     sort_keys=True),
+                          match_type,
+                          attachment_type=allure.attachment_type.JSON)
+            result = current_result
+            break
+        time.sleep(1)
+
+    assert len(result) > 0, \
+        f"The findings of type {match_type} not found"
+
+
+@pytest.mark.pre_merge
+@pytest.mark.order(4)
+@pytest.mark.dependency(depends=["test_cloudbeat_pod_exist"])
+def test_leader_election(fixture_data, elastic_client):
+    """
+    This test verifies that findings of all types are sending to elasticsearch
+    :param elastic_client: Elastic API client
+    :return:
+    """
+
+    query, sort = build_es_query("k8s_object")
+    pods, nodes, leader_node = fixture_data
+    assert leader_node != "", \
+        f"The Leader node could not be found"
+
+    time.sleep(CONFIG_TIMEOUT)  # Wait for all pods to send resources to ES
```

Comment:
I'd like to prevent usage of time.sleep. Can we implement more elegant solution?

Reply:
I would prefer a different solution but I couldn't think of a way to verify that all agents finished sending their findings.
We can check that an agent sent at least a single finding but it would not ensure that all of them were sent.

Reply:
Sure, but it won't ensure you that all agents finished sending their findings (k8s-related findings in particular).
Having said that, it might be sufficient.